pkg/controller/controllercmd/builder.go

```diff
@@ -70,6 +73,21 @@ func (b *ControllerBuilder) WithFileObserver(reactorFunc func(file string, actio
 	return b
 }
 
+func (b *ControllerBuilder) WithEventRecorder() *ControllerBuilder {
+	clientConfig, err := b.getClientConfig()
+	if err != nil {
+		panic(fmt.Sprintf("error getting client config: %v", err))
+	}
+	kubeClient := kubernetes.NewForConfigOrDie(clientConfig)
+	replicaSetRef, err := events.GetReplicaSetOwnerReference(kubeClient.AppsV1().ReplicaSets(b.componentName))
```

Comment:
doesn't this force us to be managed by ReplicaSet i.e. Deployments? What about DaemonSets? This does not look generic.

Reply:
Would expect a way to inject the way the target object is derived, with a default implementation for replicasets.

Reply:
@stts to support daemonsets we will have to bake that logic here, otherwise we will be bloating the NewCommand? Since we only will support deployments and daemonsets, I think we can do a simple if here based on the existence of environment variable? (if REPLICASET_POD or DAEMONSET_POD ?)

Reply:
The current implementation is generic, this was just bad name. It handles every pod that has controller owner ref (it follow that owner ref to get the controller object (replicaset/daemonset)